### PR TITLE
feat(HMS-1429): Add GCP to spec

### PR DESCRIPTION
**Pull request by adiabramovitch**
This PR depends on #468, and adds GCP to spec

### CODE REVIEW

**Comment by lzap** [comment]
Please add examples just like in: https://github.com/RHEnVision/provisioning-backend/pull/557

(Perhaps wait until that one is merged and rebase on top of that.)

File: docs/configure-gcp.md
Comment (adiabramovitch) [reply docs/configure-gcp.md:63]: yes, I forgot to put this PR on draft, thanks!

**Comment by akhil-jha** [comment]
@adiabramovitch Just tag me when this PR is ready :) I'd want to get started.

File: docs/configure-gcp.md
Comment (mshriver) [reply docs/configure-gcp.md:63]: 50% making sure I understand the setup! :)

**Comment by adiabramovitch** [comment]
> @adiabramovitch Just tag me when this PR is ready :) I'd want to get started.

@akhil-jha I think you can start :)

**Comment by akhil-jha** [comment]
Thanks. I think you're missing for GCP. 
```
        "tags": [
          "Reservation"
        ]
```

**Comment by adiabramovitch** [comment]
> Thanks. I think you're missing for GCP.
> 
> ```
>         "tags": [
>           "Reservation"
>         ]
> ```

Right, updated :)

File: cmd/spec/example_reservation.go
Comment (lzap) [inline cmd/spec/example_reservation.go:191]: Numbers seem to be a bit longer, `5293983267767892774` in my case, but ok :-)

File: cmd/spec/example_reservation.go
Comment (lzap) [inline cmd/spec/example_reservation.go:174]: Could you provide example for the operation name and maybe also template name? If we ever commit and push this, we will definitely forget to fill that in when the feature finally lands. Also, operation name I don’t remember what this is for, maybe the example can help.

File: cmd/spec/path.yaml
Comment (lzap) [inline cmd/spec/path.yaml:476]: Please explain the name pattern here, we do not drop a word about it.

File: internal/payloads/reservation_payload.go
Comment (lzap) [inline internal/payloads/reservation_payload.go:224]: This should be refactored, creating different JSON names and Go field type names is something we should definitely avoid.

```suggestion
	NamePattern string `json:"name_pattern" yaml:"name_pattern"`
```

It is fine to keep this as `Name` in the response as long as the JSON/YAML has the same name. But for request, I think it is more appropriate to name it as a pattern.

**Review by lzap — CHANGES_REQUESTED** [review]
Hey, I have some comments.

We should talk on tech discussion on the name pattern precedent, is this something we want to apply for aws/azure too?

File: internal/payloads/reservation_payload.go
Comment (adiabramovitch) [reply internal/payloads/reservation_payload.go:224]: So regarding the Name pattern, in a bulkInsert Request, we do not have a name field but a name pattern field which needs to have the ### sign. 
I will change the GCP naming from Name to NamePattern

Comment (lzap) [reply internal/payloads/reservation_payload.go]: Yeah I get that, it is okay to have Name in response but NamePattern in request. It seems logical and correct to me :-)

File: api/openapi.gen.json
Comment (lzap) [inline api/openapi.gen.json:150]: Interesting so it is some sort of unique id of some backend operation...

**Review by lzap — APPROVED** [review]
Great, let’s merge when IQE is ready.

**Review by lzap — APPROVED** [review]
Yeah this will fix the `nil` issue with steps.

**Review by akhil-jha — APPROVED** [review]

**Pull request merged**
